### DISCUSSION TITLE
App check duplicates prior to uploading

### DISCUSSION THREAD

**Discussion by mathwizmf2009** [immich-app/immich#5106] [Q&A]
Is it possible for the duplicate detection to occur on the app (I'm using android) without needing to upload the file? My situation may be unique, but I recently needed in reinstall my app. In doing so, it lost all the knowledge of what was uploaded already, what could be skipped, etc. So now it's going through the painful process of re-uploading ~7k files just to figure out that they're already on the server and mark them as duplicates within the app. This doesn't feel very efficient, but maybe it's more efficient to do it this way rather than performing the duplicate calculations locally on my phone?

**Comment by mathwizmf2009** [comment]
For reference, I've used immich-go tool to bulk upload images from Google Takeout archives. That checks for possible duplicates prior to uploading using filename then cross- checking the server assets returned based on date taken and image size to determine if the local file should be uploaded. 

https://github.com/simulot/immich-go/blob/main/cmdupload/upload.go
Function: ShouldUpload.

Not sure if this is an ideal way to do this or not, hence the question rather than suggestion or feature / improvement request.

**Comment by F1LIP3** [comment]
I guess if the server could hold a hash key list of all the files the server holds, the user-end points, i.e web, android or whatever, could possibly calculate the file hash-key prior to the upload action, and ask the server if that hash-key is already there. If yes, dont upload, otherwise, upload.
